python/pyarrow/_dataset.pyx

```diff
@@ -1361,6 +1369,38 @@ cdef class ScalarExpression(Expression):
         Expression.init(self, sp)
         self.scalar = <CScalarExpression*> sp.get()
 
+    @property
+    def value(self):
+        cdef:
+            shared_ptr[CScalar] scalar = self.scalar.value()
+            DataType typ = pyarrow_wrap_data_type(scalar.get().type)
+            c_string val
+
+        if is_null(typ):
+            return None
+
+        val = scalar.get().ToString()
```

Comment:
I'm not proud of this, but this will work until we properly expose the C++ Scalar <=> Python value conversions.

Reply:
if you're going to use the string repr here then please pickle it explicitly alongside the scalar's type, then use Scalar::Parse to unpickle

Reply:
Removed in favor of the ScalarValue wrappers.